stdlib/json.gr

```diff
@@ -0,0 +1,2103 @@
+    // which is difficult and suboptimal when using a function to iterate
+    // the input string. So we don't want to pay the price in other cases.
+    // This cannot be done just in the emitCodePoint function.
+    // It could be possible to implement more optimally, but would
+    // complicate things even more than this.
+    (s: String) => {
+      Buffer.addChar('"', buffer)
+
+      let mut prevCodePoint = 0
+
+      String.forEachCodePoint(codePoint => {
+        if (codePoint == 47) {
+          if (prevCodePoint == 60) {
+            Buffer.addChar('\\', buffer)
+            Buffer.addChar('/', buffer)
+          } else {
+            // otherwise just emit the slash as-is
+            Buffer.addChar('/', buffer)
+          }
+        } else {
+          emitCodePoint(codePoint)
+        }
+
+        prevCodePoint = codePoint
+      }, s)
+
+      Buffer.addChar('"', buffer)
+    }
+  }
+
+  let implHelper = {
+    format,
+    buffer,
+    emitEscapedQuotedStringPretty: emitEscapedQuotedString,
+    printNewLine,
+    printIndentation,
+  }: JsonWriterConfig
+
+  {
+    emit: json => {
+      match (printElementPretty(json, implHelper, 0)) {
+        None => void,
+        err => return err,
+      }
+      if (format.finishWithNewLine) {
+        match (printNewLine) {
+          Some(printNewLine) => printNewLine(),
+          None => void,
+        }
+      }
+      return None
+    },
+  }: JsonWriter
+}
+
+/**
+ * Prints the JSON object into a string with specific formatting settings.
+ *
+ * @param format: formatting option
+ * @param json: The JSON object to print
+ * @returns A `Result` object with either the string containing the printed JSON or an error if the input object cannot be represented in the JSON format.
+ *
+ * @example
+ * print(
+ *   toString(
+ *     format=Custom{
+ *      indentation: NoIndentation,
+ *      arrayFormat: CompactArrayEntries,
+ *      objectFormat: CompactObjectEntries,
+ *      lineEnding: NoLineEnding,
+ *      finishWithNewLine: false,
+ *      escapeAllControlPoints: true,
+ *      escapeHTMLUnsafeSequences: true,
+ *      escapeNonASCII: true,
+ *     },
+ *     JsonObject([("currency", JsonString("€")), ("price", JsonNumber(99.9))])
+ *   )
+ * )
+ * // Output: Ok("{\"currency\":\"\\u20ac\",\"price\":99.9}")
+ * @example
+ * print(
+ *   toString(
+ *     JsonObject([("currency", JsonString("€")), ("price", JsonNumber(99.9))]
+ *   )
+ * )
+ * // Output: Ok("{\"currency\":\"€\",\"price\":99.9}")
+ * @example
+ * print(
+ *   toString(
+ *     format=Compact
+ *     JsonObject([("currency", JsonString("€")), ("price", JsonNumber(99.9))])
+ *   )
+ * )
+ * // Output: Ok("{\"currency\":\"€\",\"price\":99.9}")
+ * @example
+ * print(
+ *   toString(
+ *     format=Pretty,
+ *     JsonObject([("currency", JsonString("€")), ("price", JsonNumber(99.9))])
+ *   )
+ * )
+ * // Output: Ok("{
+ * // \"currency\": \"€\",
+ * // \"price\": 99.9
+ * //}")
```

Comment:
@spotandjake are you going to update the examples like you did with other modules in stdlib?